zerver/lib/webhooks/common.py

```diff
@@ -273,3 +277,33 @@ def parse_multipart_string(body: str) -> dict[str, str]:
         data[field_name] = body
 
     return data
+
+
+def validate_webhook_signature(
+    request: HttpRequest, payload: str, signature: str, algorithm: str = "sha256"
+) -> None:
+    if not settings.VERIFY_WEBHOOK_SIGNATURES:
+        return
+    if algorithm not in hashlib.algorithms_available:
+        raise JsonableError(
+            _("The algorithm '{algorithm}' is not supported.").format(algorithm=algorithm)
```

Comment:
This should be an `AssertionError`, since it suggests a bug in the code calling this, not in the data sent to the webhook.